code/modules/nano/nanoui.dm

```diff
@@ -432,11 +433,19 @@ nanoui is used to open and update nano browser uis
 		return // Will be closed by update_status().
 
 	show_browser(user, get_html(), "window=[window_id];[window_size][window_options]")
+	winset(user, window_id, "on-close=\"nanoclose \ref[src]\"")
 	winset(user, "mapwindow.map", "focus=true") // return keyboard focus to map
-	on_close_winset()
-	//onclose(user, window_id)
 	SSnano.ui_opened(src)
 
+ /**
+  * Move window up front.
+  *
+  * @return nothing
+  */
+/datum/nanoui/proc/focus()
+	winset(user, window_id, "focus=true")
+	winset(user, "mapwindow.map", "focus=true") // return keyboard focus to map
```

Comment:
Testing this winset command in other contexts led me to the conclusion that it's inconsistent at best and does nothing at worst. Not opposed to it exactly, but I have doubts that this focuses reliably in an actual server deployment.